app/manager/models.py

```diff
@@ -18,163 +19,126 @@ def __str__(self):
-        """String for representing"""
         return self.name
 
 
-class HostType(models.Model):
-    """Model class for defining host types"""
-
-    # Fields
-    name = models.CharField(max_length=64, null=False)
-
-    # Metadata
-    class Meta: 
-        # Default ordering
-        ordering = ['name']
+class Datacenter(models.Model):
+    name = models.CharField(max_length=64, null=False, blank=False, unique=True)
+    description = models.TextField(max_length=500, null=False, blank=False)
 
     def __str__(self):
-        """String for representing"""
         return self.name
 
 
-class HostStatus(models.Model):
-    """Model class for defining host status"""
-
-    # Fields
-    name = models.CharField(max_length=64, null=False)
-
-    # Metadata
-    class Meta: 
-        # Default ordering
-        ordering = ['name']
+class DatacenterRoom(models.Model):
+    center = models.ForeignKey(Datacenter, null=False, blank=False, on_delete=models.PROTECT)
+    name = models.CharField(max_length=64, null=False, blank=False)
+    description = models.TextField(max_length=500, null=False, blank=False)
 
     def __str__(self):
-        """String for representing"""
-        return self.name
+        return self.center.__str__() + " : " + self.name
 
+    class Meta:
+        unique_together = [['center', 'name']]
 
-class Service(models.Model):
-    """Model class for defining a service"""
 
-    # Fields
-    name = models.CharField(max_length=64, null=False)
-
-    # Metadata
-    class Meta: 
-        # Default ordering
-        ordering = ['name']
+class Cluster(models.Model):
+    room = models.ForeignKey(DatacenterRoom, null=False, blank=False, on_delete=models.PROTECT)
+    name = models.CharField(max_length=64, null=False, blank=False)
+    environment = models.ForeignKey(Environment, null=False, blank=False, on_delete=models.PROTECT)
 
     def __str__(self):
-        """String for representing"""
-        return self.name
+        return self.room.__str__() + " : " + self.environment.__str__() + " : " + self.name
 
+    class Meta:
+        unique_together = [['room', 'name', 'environment']]
 
-class Server(models.Model):
-    """Model class for defining a server"""
 
-    # Fields
-    name = models.CharField(max_length=64, null=False)
-    cpu = models.CharField(max_length=64, null=False)
-    memory = models.PositiveIntegerField()
-    u_size = models.PositiveSmallIntegerField(default=1)
-    end_support = models.DateField()
-
-    # Metadata
-    class Meta: 
-        # Default ordering
-        ordering = ['name']
+class HostType(models.Model):
+    name = models.CharField(max_length=64, null=False, blank=False, unique=True)
 
     def __str__(self):
-        """String for representing"""
         return self.name
 
 
-class Cluster(models.Model):
-    """Model class for defining a cluster"""
+class HostStatus(models.Model):
+    name = models.CharField(max_length=64, null=False, blank=False, unique=True)
```

Comment:
could be interesting to add a description field like in the other Class for reference Models to explain the status

Reply:
Done :)